cmd/libs/go2idl/client-gen/generators/generator_for_group.go

```diff
@@ -49,7 +50,11 @@ func (g *genGroup) Namers(c *generator.Context) namer.NameSystems {
 }
 
 func (g *genGroup) Imports(c *generator.Context) (imports []string) {
-	return g.imports.ImportLines()
+	imports = append(imports, g.imports.ImportLines()...)
+	if g.version != "unversioned" {
+		imports = append(imports, "fmt")
```

Comment:
goimports is probably acceptable for adding "fmt", but we have a mechanism
to get the imports exactly right for use in cases where goimports can't be
expected to figure it out, so we might as well use it for this, too.

Also I don't think the tool runs goimports unless someone added it when I
wasn't looking.

On Mon, Apr 18, 2016 at 2:46 PM, krousey <notifications@github.com> wrote:

> In cmd/libs/go2idl/client-gen/generators/generator_for_group.go
> <https://github.com/kubernetes/kubernetes/pull/24395#discussion_r60140095>
> :
>
> > @@ -49,7 +50,11 @@ func (g *genGroup) Namers(c *generator.Context) namer.NameSystems {
> >  }
> >
> >  func (g *genGroup) Imports(c *generator.Context) (imports []string) {
> > -	return g.imports.ImportLines()
> > +	imports = append(imports, g.imports.ImportLines()...)
> > +	if g.version != "unversioned" {
> > +		imports = append(imports, "fmt")
>
> I always wondered about this. Does goimports not obviate the need for this?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24395/files/7ce65c848a7a9ed4dc03eaeae0ed8e244f5a79d4#r60140095>
>